### PR TITLE
Add base rust version to the output of cargo-build-sbf --version

### PR DESCRIPTION
**Pull request by dmakarov**
#### Problem

Platform tools version is not very informative. Additional information about the version of Rust the tools are based on could be helpful.

#### Summary of Changes

Add a function that extracts the version of the rust compiler installed in the platform-tools and output the version for the `--version` command line option of cargo-build-sbf.

### CODE REVIEW

File: sdk/cargo-build-sbf/src/main.rs [hunk 1]
```diff
@@ -894,10 +908,12 @@ fn main() {
     // The following line is scanned by CI configuration script to
     // separate cargo caches according to the version of platform-tools.
     let platform_tools_version = String::from("v1.37");
+    let rust_base_version = get_base_rust_version(platform_tools_version.as_str());
     let version = format!(
-        "{}\nplatform-tools {}",
+        "{}\nplatform-tools {}{}",
```
Comment (mvines) [inline sdk/cargo-build-sbf/src/main.rs:913]: how about this instead:
```suggestion
        "{}\nplatform-tools {}\n{}",
```
so we get
```
solana-cargo-build-sbf 1.16.1
platform-tools v1.37
rustc 1.68.0-dev
```
which'll be much easier to incorporate into my PR at https://github.com/solana-labs/solana/pull/32232

File: sdk/cargo-build-sbf/src/main.rs [hunk 2]
```diff
@@ -162,6 +162,20 @@ fn get_latest_platform_tools_version() -> Result<String, String> {
...
 
+fn get_base_rust_version(platform_tools_version: &str) -> String {
+    let target_path = make_platform_tools_path_for_version("platform-tools", platform_tools_version);
+    let rustc = target_path.join("rust").join("bin").join("rustc");
+    let args = vec!["--version"];
+    let output = spawn(&rustc, args, false);
+    let rustc_re = Regex::new(r"rustc ([0-9]+\.[0-9]+\.[0-9]+).*").unwrap();
+    if rustc_re.is_match(output.as_str()) {
+        let captures = rustc_re.captures(output.as_str()).unwrap();
+        format!(", based on rust {}", &captures[1])
```
Comment (mvines) [inline sdk/cargo-build-sbf/src/main.rs:173]:
```suggestion
        format!("{}", &captures[1])
```

File: sdk/cargo-build-sbf/src/main.rs [hunk 3]
```diff
@@ -162,6 +162,20 @@ fn get_latest_platform_tools_version() -> Result<String, String> {
     Ok(version)
 }
 
+fn get_base_rust_version(platform_tools_version: &str) -> String {
+    let target_path = make_platform_tools_path_for_version("platform-tools", platform_tools_version);
+    let rustc = target_path.join("rust").join("bin").join("rustc");
+    let args = vec!["--version"];
+    let output = spawn(&rustc, args, false);
+    let rustc_re = Regex::new(r"rustc ([0-9]+\.[0-9]+\.[0-9]+).*").unwrap();
```
Comment (mvines) [inline sdk/cargo-build-sbf/src/main.rs:170]:
```suggestion
    let rustc_re = Regex::new(r"(rustc [0-9]+\.[0-9]+\.[0-9]+).*").unwrap();
```

**Comment by dmakarov** [comment]
Looks like it's ready to be merged.

**Review by mvines — APPROVED** [review]

**Pull request merged**